src/types/request.rs

```diff
@@ -0,0 +1,63 @@
+use std::{str::FromStr, time::Duration};
+
+use subsquid_messages::Range;
+
+use super::DatasetId;
+
+#[derive(Debug, Clone)]
+pub struct ClientRequest {
+    pub dataset_id: DatasetId,
+    pub query: ParsedQuery,
+    pub buffer_size: usize,
```

Comment:
Everything below is definitely something not to be exposed even in the "secret mode".

Reply:
While it's primarily designed for debugging, I don't see a reason to not allow users of their own gateway to set those parameters per request. They can set them in the config anyway.
But if we will host a public gateway, those should definitely be limited or removed at all.

Reply:
>I don't see a reason to not allow users of their own gateway to set those parameters per request

It is not polite to expose to users low level details, that could change anytime. Also, I believe, those kinds of settings could be set once and for all.

Reply:
Okay, I will hide it from public API, but I don't want to hardcode it because it slows down the development process if you have to recompile and restart the gateway (it's more than a minute) to test a new value